.github/workflows/ci-aot.yml

```diff
@@ -0,0 +1,33 @@
+name: Publish AOTCompatibility testApp
+
+on:
+  push:
+    branches: [ 'main*' ]
+    paths-ignore:
+    - '**.md'
+  pull_request:
+    branches: [ 'main*' ]
+    paths-ignore:
+    - '**.md'
+
+jobs:
+  aot-test:
+    strategy:
+      fail-fast: false  # ensures the entire test matrix is run, even if one permutation fails
+      matrix:
+        os: [ windows-latest ]
+        version: [ net6.0, net7.0 ]
```

Comment:
Why are we adding `net6.0` now?

Reply:
I don't think we need it now - AOT will run on 7.0 only anyway.
As a subsequent change adding a trim test should target 6.0 though (since that's where we introduced trimming), but we might run into similar issues as with AOT on 7, that not all of the dependent libraries are correctly annotated for trimming in 6 (specifically ASP.NET comes in mind).

Reply:
> As a subsequent change adding a trim test should target 6.0 though (since that's where we introduced trimming)

For now, could we just test both trimming and AOT on net7.0? Or would trimming be better tested on net6.0?

Reply:
I think it's a great start to run on .NET 7 always - in fact once we can I would even say run on .NET 8 only.
Technically it would be nice to support trimming at least for a subset of OTel (probably without ASP.NET) on .NET 6, but it's much lower priority, so we could easily wait for feedback on that one.

Reply:
removed net6.0 from the target for now as suggested